recipes/mailio/all/conanfile.py

```diff
@@ -27,61 +27,38 @@ class MailioConan(ConanFile):
         "shared": False,
     }
     short_paths = True
-
-    @property
-    def _min_cppstd(self):
-        return "17"
-
-    @property
-    def _compilers_minimum_version(self):
-        return {
-            "gcc": "8.3",
-            "clang": "6",
-            "Visual Studio": "15",
-            "msvc": "191",
-            "apple-clang": "10",
-        }
+    implements = ["auto_shared_fpic"]
 
     def export_sources(self):
         export_conandata_patches(self)
 
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
-
-    def configure(self):
-        if self.options.shared:
-            self.options.rm_safe("fPIC")
-
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.82.0", transitive_headers=True, transitive_libs=True)
+        self.requires("boost/1.86.0", transitive_headers=True, transitive_libs=True)
         self.requires("openssl/[>=1.1 <4]", transitive_headers=True, transitive_libs=True)
 
     def validate(self):
-        if self.settings.get_safe("compiler.cppstd"):
-            check_min_cppstd(self, self._min_cppstd)
-
-        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(
-                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-            )
+        check_min_cppstd(self, 17)
 
     def build_requirements(self):
         self.tool_requires("cmake/[>=3.16.3 <4]")
+        if Version(self.version) >= "0.24.0" and \
```

Comment:
I believe this needs to be in a `validate()` or `validate_build()` method?

Reply:
However, if the error was this:

```
/Users/jenkins/workspace/cci_prod_PR-25955/conan-home/p/b/maili77426e52e76ce/b/src/src/percent.cpp:37:25: error: implicit instantiation of undefined template 'std::vector<std::basic_string<char>>'
vector<string> percent::encode(const string& txt, const string& charset) const
                        ^
/Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iosfwd:216:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
/Users/jenkins/workspace/cci_prod_PR-25955/conan-home/p/b/maili77426e52e76ce/b/src/src/percent.cpp:39:20: error: implicit instantiation of undefined template 'std::vector<std::basic_string<char>>'
    vector<string> enc_text;
                   ^
/Applications/conan/xcode/13.0/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/include/c++/v1/iosfwd:216:28: note: template is declared here
class _LIBCPP_TEMPLATE_VIS vector;
                           ^
```

there's a chance the fix is just including `<vector>` in the right place

Reply:
The [version 0.24.1](https://github.com/karastojko/mailio/releases/tag/0.24.1) with the proposed fix ix ready.

Reply:
@karastojko 
Thanks a lot!